### ISSUE TITLE
how to execute for flavors in gradle?

### ISSUE THREAD

**Issue by wintersandroid** [avianey/androidsvgdrawable-plugin#23]
I have a project with multiple flavors and a main - most of the svg files live in main, but the main app icons are in each flavour

I create a configuration for each flavour and the main, but the plugin executes for all flavours not just the one being built..

Is there a way with this plugin to link each configuration to a flavour?

**Comment by avianey** [comment]
In your gradle build script, remove the apply plugin declaration :
```
apply plugin: "fr.avianey.androidsvgdrawable"
```
Then you can define a tasks to be executed for the desired variant only :
```
task svgToPng<Product Flavor Name>(type: fr.avianey.androidsvgdrawable.gradle.SvgDrawableTask) {
   // ...
}
project.afterEvaluate {
    assemble<Product Flavor Name>.dependsOn('svgDrawable')
}
```
See http://tools.android.com/tech-docs/new-build-system/user-guide#TOC-Building-and-Tasks to bind the task correctly. The problem is that the plugin automatically binds `SvgDrawableTask` to the Android `preBuild` task... that is the same for every flavour.

Or you can try this : http://stackoverflow.com/a/22743738/1638277 still removing apply plugin declaration.

Tell me if it works for you.

**Comment by wintersandroid** [comment]
I had a go, could not get it to work.. 
Seemed to fail to find 'svgDrawable'

**Comment by avianey** [comment]
At this point I have to solution (and I think both will be implemented) :
# have a new parameter to specify the task to bind to (each SvgDrawableTask could be bound to anything, 'preBuild' will stay the default)
# have a new parameter to say to the SvgDrawable plugin not to execute a given SvgDrawableTask automatically... this task should be executed by an explicit closure in the build file

Working on it...
This might fix #24 by specifying  something else than `preBuild`

**Comment by avianey** [comment]
Documented here :
https://github.com/avianey/androidsvgdrawable-plugin/wiki/How-to-use-with-flavor
